src/rdb/rdb_raft.c

```diff
@@ -230,25 +229,25 @@ rdb_lc_store_replicas(daos_handle_t lc, uint64_t index,
 }
 
 static int
-rdb_raft_add_node(struct rdb *db, d_rank_t rank, bool is_self, bool is_voting)
+rdb_raft_add_node(struct rdb *db, d_rank_t rank)
 {
 	struct rdb_raft_node	*dnode;
 	raft_node_t		*node;
+	d_rank_t		 self;
 	int			 rc = 0;
 
 	D_ALLOC_PTR(dnode);
 	if (dnode == NULL)
 		D_GOTO(out, rc = -DER_NOMEM);
+	rc = crt_group_rank(NULL, &self);
+	D_ASSERTF(rc == 0, "%d\n", rc);
 	dnode->dn_rank = rank;
-	node = raft_add_node(db->d_raft, dnode, rank, is_self);
+	node = raft_add_node(db->d_raft, dnode, rank, (rank == self));
 	if (node == NULL) {
 		D_ERROR(DF_DB": failed to add node %u\n", DP_DB(db), rank);
 		D_FREE_PTR(dnode);
```

Comment:
Just a note, a few D_FREE_PTR's have crept back into the code.  Can just use D_FREE as D_FREE_PTR adds no value